### PR TITLE
[WIP] Bring PV Camera support back (+ DomU changes)

### PR DESCRIPTION
**Pull request by otyshchenko1**
TODO:
1. PV Camera support shouldn't be enabled by default, but under config option or system should 
    remain functional if no real Camera attached at least
2. Patches to domain config files could be squashed
3. Need a decision regarding USB Camera support

The completely new patch in this series is "Add vspfilter rules and update vspfilter configs to match them"

### CODE REVIEW

**Comment by otyshchenko1** [comment]
The corresponding PRs: https://github.com/xen-troops/xen/pull/159 and https://github.com/xen-troops/linux/pulls

**Comment by otyshchenko1** [comment]
Update by adding "pvcamera" distro-feature

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/wayland/weston/camera_front.rules [hunk 1]
```diff
@@ -0,0 +1 @@
+SUBSYSTEM=="video4linux", ENV{ID_BUS}=="usb", ENV{ID_V4L_CAPABILITIES}=="*capture*", SYMLINK+="cam_front"
```
Comment (rshym) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/wayland/weston/camera_front.rules]: May be it's better to name file camera_usb.rules, not camera_front.rules.
And same for symlink - maybe just 'camera'.
On one side we have to take into account other types of camera. But also we can make assumption that only one type is present in system. So need to discuss.

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-core/packagegroups/packagegroup-xt-core-pv.bb [hunk 2]
```diff
@@ -8,4 +8,5 @@ RDEPENDS_packagegroup-xt-core-pv = "\
     libxenbe \
     displbe \
     sndbe \
+    camerabe \
```
Comment (rshym) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-core/packagegroups/packagegroup-xt-core-pv.bb]: May it should be wrapped by feature?

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8.bbappend [hunk 3]
```diff
@@ -5,7 +5,7 @@ SRCREV = "${AUTOREV}"
 LINUX_VERSION = "4.14.75"
 
 SRC_URI = " \
-    git://github.com/xen-troops/linux.git;branch=${BRANCH} \
+    git://github.com/otyshchenko1/linux.git;branch=${BRANCH} \
```
Comment (arminn) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8.bbappend]: Is it ok to use personal repo?

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/camerabe/camerabe_git.bbappend [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+SYSTEMD_SERVICE_${PN} = "camerabe.service"
+
+do_install_append() {
+    install -d ${D}${systemd_system_unitdir}
+    install -m 0644 ${WORKDIR}/camerabe.service ${D}${systemd_system_unitdir}
+}
+
+FILES_${PN} += " \
+    ${systemd_system_unitdir}/camerabe.service \
+"
```
Comment (arminn) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/camerabe/camerabe_git.bbappend]: No newline ....

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8.bbappend (hunk 3 above)
Comment (otyshchenko1) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8.bbappend]: That patch is not for merge as it's subject actually claims, but for possibility to build with required changes to Linux applied and minimal integration effort. This patch will be droped when corresponding PR to Linux goes in.

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/camerabe/camerabe_git.bbappend (hunk 4 above)
Comment (otyshchenko1) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/camerabe/camerabe_git.bbappend]: Indeed, will add

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-core/packagegroups/packagegroup-xt-core-pv.bb (hunk 2 above)
Comment (otyshchenko1) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-core/packagegroups/packagegroup-xt-core-pv.bb]: Yes, next patch does that https://github.com/xen-troops/meta-xt-prod-devel/pull/246/commits/9e91aa4665c9cd1a8bab469f37bfd13aa35b847a#diff-fe7cf7d393e7b827edb506c882f7a8ddR11

Comment (rshym) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-core/packagegroups/packagegroup-xt-core-pv.bb]: Thanks. I missed it.

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/wayland/weston/camera_front.rules (hunk 1 above)
Comment (otyshchenko1) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/wayland/weston/camera_front.rules]: Good point. I was thinking about it. If you ok, we will certainly rename it accordingly in the follow up series when adding support for other camera types. Currently in the context of restoring pv camera support task, I propose to keep it unchanged.

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8.bbappend (hunk 3 above)
Comment (otyshchenko1) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8.bbappend]: Already merged https://github.com/xen-troops/linux/pull/81 and updated current PR by removing a commit in question.

**Comment by otyshchenko1** [comment]
TODO:

~~1. PV Camera support shouldn't be enabled by default, but under config option or system should
remain functional if no real Camera attached at least~~
~~2. Patches to domain config files could be squashed~~
3. Need a decision regarding USB Camera support

**Comment by otyshchenko1** [comment]
Updated by removing last patch that uncomments #DISTRO_FEATURES_append = " pvcamera". So PV camera remains disabled by default.

**Pull request merged**